service/service_account.go

```diff
@@ -64,14 +64,55 @@ func (s AccountService) AccountBalance(
 		return nil, wrapError(errInternalError, metadataErr)
 	}
 
+	avaxBalance, balanceErr := s.client.BalanceAt(context.Background(), address, header.Number)
+	if balanceErr != nil {
+		return nil, wrapError(errInternalError, balanceErr)
+	}
+
+	var balances []*types.Amount
+	balances = append(balances, mapper.AvaxAmount(avaxBalance))
+	for _, currency := range req.Currencies {
+		value, ok := currency.Metadata[mapper.ContractAddressMetadata]
+		if !ok {
+			if currency.Decimals == AvaxDecimals && strings.ToLower(currency.Symbol) == AvaxSymbolLower {
```

Comment:
This is not the right match check for a currency input.

Reply:
You should use `types.Hash` here: https://github.com/coinbase/rosetta-sdk-go/blob/2868f1040ba0db56acedd72d2ec745d0c2c44d57/reconciler/reconciler.go#L181-L184